tests/libvmifact/factory.go

```diff
@@ -146,3 +148,23 @@ func NewWindows(opts ...libvmi.Option) *kvirtv1.VirtualMachineInstance {
 	vmi.Spec.Domain.Firmware = &kvirtv1.Firmware{UUID: WindowsFirmware}
 	return vmi
 }
+
+func NewTinyOSVMWithDataVolume(dv *cdiv1.DataVolume, extraVMOpts ...libvmi.VMOption) *kvirtv1.VirtualMachine {
+	// Supplied with no user data, AlpimeWithTestTooling image takes more than 200s to allow login
+	withNonEmptyUserData := libvmi.WithCloudInitNoCloudEncodedUserData("#!/bin/bash\necho hello\n")
+	opts := []libvmi.VMOption{
+		libvmi.WithDataVolumeTemplate(dv),
+	}
+	opts = append(opts, extraVMOpts...)
+
+	return libvmi.NewVirtualMachine(
+		libvmi.New(
+			libvmi.WithDataVolume("disk0", dv.Name),
+			libvmi.WithResourceMemory(cirrosMemory()),
```

Comment:
This option is already part of the `Guestless` factory. You could use it.

Reply:
minimum qemu memory != minimum memory for cirros and the like
https://github.com/kubevirt/kubevirt/pull/11366#issuecomment-1970247448